File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/domain/usecase/ConnectDeviceUseCase.kt
package com.maktubcompany.easy_blue_printer.plugin.domain.usecase


import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository

class ConnectDeviceUseCase(private val repository: BluetoothRepository) {
    fun execute(address: String): Boolean {
        return repository.connectToDevice(address)
    }
}

File: ios/easy_blue_printer/Sources/easy_blue_printer/plugin/domain/usecases/ConnectDeviceUseCase.swift
import Foundation

public class ConnectDeviceUseCase {
    private let repository: BluetoothRepository

    public init(repository: BluetoothRepository) {
        self.repository = repository
    }

    public func execute(address: String) -> Bool {
        return repository.connectToDevice(address: address)
    }
}
